code/modules/supply/supply_console.dm

```diff
@@ -0,0 +1,499 @@
+		return
+
+	. = TRUE
+	add_fingerprint(user)
+
+	switch(action)
+		if("moveShuttle")
+			move_shuttle(user)
+		if("order")
+			if(world.time < reqtime)
+				visible_message("<b>[src]</b>'s monitor flashes, \"[world.time - reqtime] seconds remaining until another requisition form may be printed.\"")
+				return
+			var/amount = 1
+			if(params["multiple"] == "1") // 1 is a string here. DO NOT MAKE THIS A BOOLEAN YOU DORK
+				var/num_input = input(user, "Amount", "How many crates? (20 Max)") as null|num
+				if(!num_input || (!is_public && !is_authorized(user)) || ..()) // Make sure they dont walk away
+					return
+				amount = clamp(round(num_input), 1, 10)
+			var/datum/supply_packs/P = locateUID(params["crate"])
+			if(!istype(P))
+				return
+			var/timeout = world.time + (60 SECONDS) // If you dont type the reason within a minute, theres bigger problems here
+			var/reason = input(user, "Reason", "Why do you require this item?","") as null|text
+			if(world.time > timeout || !reason || (!is_public && !is_authorized(user)) || ..())
+				// Cancel if they take too long, they dont give a reason, they aint authed, or if they walked away
+				return
+			reason = sanitize(copytext(reason, 1, MAX_MESSAGE_LEN))
+
+			//===orderee identification information===
+			var/idname = "*None Provided*"
+			var/idrank = "*None Provided*"
+			if(ishuman(user))
+				var/mob/living/carbon/human/H = user
+				idname = H.get_authentification_name()
+				idrank = H.get_assignment()
+			else if(issilicon(user))
+				idname = user.real_name
+
+			//===orderee account information===
+			var/datum/money_account/selected_account = locateUID(params["account"])
+			for(var/i in 1 to amount)
+				var/datum/supply_order/order = SSeconomy.generate_supply_order(params["crate"], idname, idrank, amount, reason)
+				order_crate(user, order, selected_account)
+				if(i == 1)
+					playsound(loc, 'sound/machines/ping.ogg', 15, 0)
+					to_chat(user, "<span class='notice'>Order Sent.</span>")
+					generate_requisition_paper(order)
+
+		if("approve")
+			var/ordernum = text2num(params["ordernum"])
+			if(!ordernum)
+				return
+			approve_crate(user, ordernum)
+		if("deny")
+			var/ordernum = text2num(params["ordernum"])
+			if(!ordernum)
+				return
+			deny_crate(user, ordernum)
+
+
+		// Popup to show CC message logs. Its easier this way to avoid box-spam in TGUI
+		if("showMessages")
+			// Public consoles cant view messages
+			if(is_public)
+				return
+			var/datum/browser/ccmsg_browser = new(user, "ccmsg", "Central Command Cargo Message Log", 800, 600)
+			ccmsg_browser.set_content(SSeconomy.centcom_message)
+			ccmsg_browser.open()
+
+/obj/machinery/computer/supplycomp/proc/order_crate(mob/user, datum/supply_order/order, datum/money_account/account)
+	var/datum/money_account/selected_account = account
+	if(!istype(selected_account)) //if no account found, just default to cargo account
+		CRASH("order_crate called without a specified money account")
+	else
```

Comment:
Don't need this one. `CRASH` will stop the proc and thus sorta function like a `return` in this context

Reply:
cleaned up